templates/CRM/Contact/Form/Merge.tpl

```diff
@@ -283,7 +283,7 @@
 
     // Update primary label
     if (mainBlock != false && mainBlock['is_primary'] == '1') {
-      this_controls.find(".location_primary").text('Primary');
+      this_controls.find(".location_primary").text('{/literal}{ts}Primary{/ts}{literal}');
```

Comment:
For this kind of embedded Javascript, one needs different escaping - because the translated string could have different/incompatible range of characters. (Ex: `{ts}It was good{/ts}` becomes something like `C'etait bon`; the unescaped single-quote in the translated data would muck with JS.) 

Example: `templates/CRM/Mailing/Form/InsertTokens.tpl` is a `*.tpl` file with Javascript + `{ts}`

```smarty
placeholder: '{/literal}{ts escape='js'}Tokens{/ts}{literal}'
```

Reply:
+1 for everything else

Reply:
Ah ha, of course that makes sense. I've updated the pull request to use the JS strategy as appropriate.